app/services/analytics.rb

```diff
@@ -125,6 +125,7 @@ def resolved_authn_context_result
       service_provider:,
       vtr: session[:sp][:vtr],
       acr_values: session[:sp][:acr_values],
+      saml: !!session[:sp][:request_url]&.include?('?SAMLRequest='),
```

Comment:
I don't have much context here about how the `Anlaytics` class works, and I have questions that are maybe outside the scope of this change. I'm happy to take this discussion outside of the PR if appropriate. 

That said, since the request_url should only exist when coming to this code from `ApplicationController`, is it better for `ApplicationController` to inject its `resolved_authn_context_result` instead of duplicating code here? Or is there something deliberately different here in the `Analytics` class?

Reply:
unfortunately i also don't have context on this code, so i don't have a lot of answers for you! i wasn't sure why this code was duplicated -- i assumed there was a reason the ApplicationController method isn't injected in, but didn't trace it through to make sure. that refactor felt out of scope to me, like you suggested.

Reply:
@solipet it looks like you introduced this method -- do you have more context into why the ApplicationController data wasn't more readily accessible/different?

Reply:
This is related to how and when session is initialized and decorated. There is technical debt here, but as noted, it's not in scope for this work.